packages/block-editor/src/components/copy-handler/index.js

```diff
@@ -213,6 +217,23 @@ function CopyHandler( { children } ) {
 	return <div ref={ useClipboardHandler() }>{ children }</div>;
 }
 
+/**
+ * Given a string of HTML representing serialized blocks, returns the plain
+ * text extracted after stripping the HTML of any tags and fixing line breaks.
+ *
+ * @param {string} html Serialized blocks.
+ * @return {string} The plain-text content with any html removed.
+ */
+function toPlainText( html ) {
+	// Manually handle BR tags as line breaks prior to `stripHTML` call
+	html = html.replace( /<br>/g, '\n' );
+
+	const plainText = stripHTML( html ).trim();
+
+	// Merge any consecutive line breaks
+	return plainText.replace( /\n\n+/g, '\n\n' );
+}
+
```

Comment:
I took the liberty of pushing this change to improve the produced plain text.

In my testing I was copying this partial content across two paragraphs:

> abc
>
> def **ghi** jkl
> _abcdefghi_

**Before**, the resulting plain text was:

```
abc
def ghi jklabcdefghi
```

**Now**:

```
abc

def ghi jkl
abcdefghi
```

Reply:
I grepped around the `raw-handling` menu to see if I could/should reuse anything to clean up the HTML while preserving line breaks, but didn't find anything that felt appropriate, hence this little dedicated function. (cc @ellatrix in case you have a better idea)

Reply:
Super awesome 👏🏻